src/app/child-dev-project/children/children-count-dashboard/children-count-dashboard.component.ts

```diff
@@ -1,54 +1,84 @@
-import { Component, OnInit } from "@angular/core";
+import { Component, Input, OnInit } from "@angular/core";
 import { ChildrenService } from "../children.service";
 import { Router } from "@angular/router";
-import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
 import { OnInitDynamicComponent } from "../../../core/view/dynamic-components/on-init-dynamic-component.interface";
+import { take } from "rxjs/operators";
+import { ConfigurableEnumValue } from "../../../core/configurable-enum/configurable-enum.interface";
 
-@UntilDestroy()
 @Component({
   selector: "app-children-count-dashboard",
   templateUrl: "./children-count-dashboard.component.html",
   styleUrls: ["./children-count-dashboard.component.scss"],
 })
 export class ChildrenCountDashboardComponent
   implements OnInitDynamicComponent, OnInit {
+  /**
+   * The property of the Child entities to group counts by.
+   *
+   * Default is "center".
+   */
+  @Input() groupBy: string = "center";
+
   totalChildren: number;
-  childrenByCenter = [];
+  childrenGroupCounts = [];
 
   constructor(
     private childrenService: ChildrenService,
     public router: Router
   ) {}
 
-  onInitFromDynamicConfig(config: any) {}
+  onInitFromDynamicConfig(config: any) {
+    if (config?.groupBy && config.groupBy !== this.groupBy) {
+      this.groupBy = config.groupBy;
+    }
+  }
 
   ngOnInit() {
     this.childrenService
       .getChildren()
-      .pipe(untilDestroyed(this))
+      .pipe(take(1))
       .subscribe((results) => {
         this.totalChildren = 0;
 
-        const countMap = new Map<string, number>();
+        const countMap = new Map<any, number>();
         results.forEach((child) => {
           if (child.isActive) {
-            let count = countMap.get(child.center);
+            let count = countMap.get(child[this.groupBy]);
             if (count === undefined) {
               count = 0;
             }
 
             count++;
             this.totalChildren++;
-            countMap.set(child.center, count);
+            countMap.set(child[this.groupBy], count);
           }
         });
-        this.childrenByCenter = Array.from(countMap.entries()); // direct use of Map creates change detection problems
+        this.childrenGroupCounts = Array.from(countMap.entries()) // direct use of Map creates change detection problems
+          .map((entry) => [extractHumanReadableLabel(entry[0]), entry[1]]);
```

Comment:
```suggestion
          .map((entry) => {name: extractHumanReadableLabel(entry[0]), label: entry[1]});
```
See the comment at the `childrenGroupCounts` Attribute. Instead of creating a two dimensional array create an array of objects.

Reply:
great suggestion! that makes it much safer and clearer